cmd/ocsp-updater/main.go

```diff
@@ -268,6 +269,76 @@ func (updater *OCSPUpdater) oldOCSPResponsesTick(batchSize int) {
 	updater.generateOCSPResponses(statuses)
 }
 
+func (updater *OCSPUpdater) getSerialsIssuedSince(since time.Time, batchSize int) ([]string, error) {
+	var serials []string
+	_, err := updater.dbMap.Select(
+		&serials,
+		`SELECT serial FROM certificates
+		 WHERE issued > :since
+		 LIMIT :limit`,
+		map[string]interface{}{
+			"since": since,
+			"limit": batchSize,
+		},
+	)
+	if err == sql.ErrNoRows {
+		return serials, nil
+	}
+	return serials, err
+}
+
+func (updater *OCSPUpdater) getNumberOfReceipts(serial string) (int, error) {
+	var count int
+	err := updater.dbMap.SelectOne(
+		&count,
+		"SELECT COUNT(id) FROM sctReceipts WHERE certificateSerial = :serial",
+		map[string]interface{}{"serial": serial},
+	)
+	return count, err
+}
+
+// missingReceiptsTick looks for certificates without the correct number of SCT
+// receipts and retrieves them
+func (updater *OCSPUpdater) missingReceiptsTick(batchSize int) {
+	now := updater.clk.Now()
+	since := now.Add(-updater.oldestIssuedSCT)
+	serials, err := updater.getSerialsIssuedSince(since, batchSize)
+	if err != nil {
+		updater.log.AuditErr(err)
+		return
+	}
+
+	for _, serial := range serials {
+		// TODO(rolandshoemaker): For now this will do, in the future we should
+		// really have another method that allows us to specify which log/s should
+		// be submitted to so we don't double submit to logs we already have receipts
+		// for
+		if count, err := updater.getNumberOfReceipts(serial); err == nil && count != updater.numLogs {
+			var certDER []byte
+			updater.dbMap.SelectOne(
+				&certDER,
+				`SELECT der FROM certificates
+					 WHERE serial = :serial`,
+				map[string]interface{}{"serial": serial},
+			)
+			if err != nil {
+				updater.log.AuditErr(err)
+				continue
+			}
+
+			err = updater.pubc.SubmitToCT(certDER)
+			if err != nil {
+				updater.log.AuditErr(err)
+				continue
+			}
+		} else if err != nil {
```

Comment:
The code right above here has gotten pretty nested. I think in the trade-off between them, it might be okay to add a line by moving the getNumberOfReceipts out and then handling the err != nil as the first condition with an early continue.